### DISCUSSION TITLE
Trying to get Element to use BlackHole as input in order to use plugins on system audio

### DISCUSSION THREAD

**Discussion by agentirons** [ExistentialAudio/BlackHole#767] [Q&A] [answered]
I have Element installed from [https://github.com/kushview/element](GitHub/Kushview). I am attempting to setup separate EQ for left and right channels of my Macbook Air M2 system audio, as discussed [here](https://www.reddit.com/r/headphones/comments/16254xf/comment/jxybb2i/).

<img width="1113" alt="Screenshot 2024-02-02 at 16 53 01" src="https://github.com/ExistentialAudio/BlackHole/assets/17654661/9a74620f-8551-4106-96fd-cb25147af7d6">

I have BlackHole installed, and I've attempted following the instructions both from here and Element, but I cannot seem to find the right combination of inputs and outputs to do what I want. In the above setup, I can hear audio from the Macbook speakers (using Tidal music player set to System Default output) but modifying a Volume plugin in Element doesn't have any effect. If I change the macOS system output to BlackHole, I get no sound. Using an Aggregate Device instead of a Multi-Output Device has similar issues in various combinations.

Could someone point me in the right direction? I started an issue at Element but the lead there wasn't able to figure it out either.

**Comment by devinroth** [comment]
Doesn’t sound like it should be hard.

BlackHole as system output
BlackHole as input into Element
Speakers as output from Element

Have you tried that?

**Reply by agentirons** [reply]
In that scenario I get no sound output at all.

<img width="987" alt="Screenshot 2024-02-03 at 07 00 01" src="https://github.com/ExistentialAudio/BlackHole/assets/17654661/55eb1b81-033d-4f0a-8332-79cad25fce01">

**Comment by devinroth** [comment]
Is there any indication in that scenario that you’re getting audio from BlackHole?

Does that work if you use your mic as the input?

**Comment by agentirons** [comment]
I do get a level indication from BlackHole in macOS sound settings, as I play music:

<img width="1444" alt="Screenshot 2024-02-03 at 10 51 58" src="https://github.com/ExistentialAudio/BlackHole/assets/17654661/ed8c077d-39e2-4139-b45f-72fc55d8daa3">

But I don't have any output from the speakers. Same if I set the input to mic (either in Element, or sound settings, or both.) Mic indicates levels on input, but no output.

**Comment by devinroth** [comment]
Maybe this isn’t in the capabilities of Element.

**Comment by agentirons** [comment]
Could be. Let's say I wanted to take Element out of the equation. I would install BlackHole, and then set it as my system input, with MacBook speakers as system output. Then tell an app like Tidal to output to BlackHole 2ch, at which point I should hear audio? Currently I do not.

I also followed the instructions to create a Multi-Output Device, with MacBook speakers listed first. I set the Multi as my system output and BlackHole as input, then verified in system settings that BlackHole was receiving audio - still no output audible.

What's like the bare minimum setup just so I can confirm that BlackHole is functioning as expected?

-------- Original Message --------

**Comment by devinroth** [comment]
You got signal from BlackHole already so it’s working correct.

My understanding is you want to capture audio, the sound, then hear it on your speakers. Correct?

**Comment by agentirons** [comment]
Yes, that is correct. I think I understand now that I need:
1) An application directing output to BlackHole - I'm using Tidal.
2) Some application which uses BlackHole as input and outputs to Speakers - I was using Element.

I installed Audacity to test that and sure enough if I turn on 'Audible Input Monitoring' and hit record I can hear and see the music from Tidal going from BlackHole to Audacity to the speakers. I don't quite understand what all the Multi-Output/Aggregate Device stuff is for but it seems like it's not relevant to my use anyway?

I think at least with the above I can go back to the Element maintainer to further troubleshoot.

**Comment by devinroth** [comment, marked as answer]
Correct Multi outputs aren’t relevant to your situation. It’s only if you want to capture but you want to capture and alter the audio.

Not open source but you could get this working with GarageBand.
